### PR TITLE
submitting test cases for backend #76

### LINKED ISSUE #76: Create server side testing infrastructure
**Is your feature request related to a problem? Please describe.**
Current server side code has no tests. In order to add tests, a testing infrastructure needs to be added to the project, along with test automation. 

**Describe the solution you'd like**
Pytest is a good framework often used for python testing. It works well with flask applications. Create a ```tests``` directory under the ServerCode directory. Add files with tests for the server side application. For each python file X.py in the ServerCode there should be a test_X.py in the tests code. Test the routes and the machine vision components.

Add a GitHub workflow that will automatically run all tests on each push to the repo.
Here's a sample workflow that AI generated for me, when I asked it to do this work (claude.ai):
```
name: Flask Application Tests

on:
  push:
  pull_request:

jobs:
  test:
    runs-on: ubuntu-latest
    
    steps:
    - uses: actions/checkout@v2
    
    - name: Set up Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.x'
    
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -r requirements.txt
        pip install pytest pytest-cov
    
    - name: Run tests with pytest
      run: pytest tests/ --cov=./ --cov-report=xml
    
    - name: Upload coverage to Codecov
      uses: codecov/codecov-action@v3
      with:
        file: ./coverage.xml
        flags: unittests
        name: codecov-umbrella
        fail_ci_if_error: true

    - name: Archive test results
      uses: actions/upload-artifact@v2
      with:
        name: test-results
        path: test-results/
      if: always()
```

### PR DESCRIPTION
**Pull request by yildizahmetali**
Fixes #76 

What was changed?

- Added validation logic in the /api/upload route to handle invalid file types and return appropriate status codes (400 for invalid files).
- Fixed the /api/process_pdf route to properly handle required parameters (pdf_file and file_id).
- Implemented proper CSV file generation and download logic in the /api/download_csv route, ensuring the correct mimetype (text/csv) is returned.
- Improved error handling in the /api/csv_acknowledgment route to return a 404 error if the file ID is not found.
- Created the /api/json_to_csv route to convert JSON data into CSV format and added error handling for malformed JSON.
- Added unit tests to ensure the API returns correct status codes, processes valid file uploads correctly, and handles errors gracefully.

Why was it changed?

- The issue was related to improper validation of file uploads, incorrect handling of parameters in the /api/process_pdf route, and missing or incomplete logic for CSV generation and acknowledgment.
- The tests were failing due to missing validation, incorrect responses (mimetype mismatches), and unhandled error cases in key routes. The changes ensure the API behaves correctly under various conditions, including invalid input and file processing errors.

How was it changed?

AppServer.py:
- Modified the /api/upload route to include file type validation and proper error handling.
- Fixed parameter handling in /api/process_pdf.
- Updated the CSV download logic in /api/download_csv to return files with the correct text/csv mimetype.
- Added error handling for invalid file IDs in /api/csv_acknowledgment.
- test_file_upload.py, test_AppServer.py, test_json_to_csv.py:
- Added tests for file type validation, JSON-to-CSV conversion, and CSV acknowledgment handling.
- Updated tests to check for proper status codes (200, 400, and 404 as appropriate).

### CODE REVIEW

**Review by yildizahmetali — COMMENTED** [review]
fixed backend tests

File: ServerCode/tests/test_AppServer.py
Comment (Chirag2x) [inline ServerCode/tests/test_AppServer.py:1]: It's good you re using comments which helps to understand.

File: ServerCode/tests/test_file_upload.py
Comment (Chirag2x) [inline ServerCode/tests/test_file_upload.py:1]: The test covers the border goals.

File: ServerCode/tests/test_json_to_csv.py
Comment (Chirag2x) [inline ServerCode/tests/test_json_to_csv.py:1]: You have to solve the part which was skipped.

**Review by Chirag2x — APPROVED** [review]
Some test cases which are skipped need to be resolved. Otherwise the test cover the border context. The structure for the test cases are good.

**Pull request merged**